frontend/src/components/pages/NewDataMapper.js

```diff
@@ -132,20 +166,23 @@ const NewDataMapper = ({ gateway, goToDataMappers }) => {
     }
   };
 
-  const selectedDatabase = glueDatabase
-    ? glueData.databases.find((x) => x.name === glueDatabase)
-    : undefined;
-
-  const selectedTable = selectedDatabase
-    ? selectedDatabase.tables.find((t) => t.name === glueTable)
-    : undefined;
+  useEffect(() => {
+    setGlueTable("-1");
+    const tableRef = document.getElementById("glueTable");
+    if (tableRef) tableRef.selectedIndex = 0;
+  }, [glueDatabase, setGlueTable]);
 
-  const tablesForSelectedDatabase = selectedDatabase
-    ? selectedDatabase.tables
-    : [];
+  useEffect(() => {
+    setColumns({ type: "reset" });
+    let checkboxes = document.getElementsByName("column");
+    for (let i = 0; i < checkboxes.length; i++) checkboxes[i].checked = false;
 
-  const columnsForSelectedTable = selectedTable ? selectedTable.columns : [];
-  const noTables = !glueData || isEmpty(glueData.databases);
+    const partitionKeys = selectedTable?.partitionKeys || [];
+    setPartitionKeys({ type: "reset", value: partitionKeys });
+    checkboxes = document.getElementsByName("partition-key");
+    for (let i = 0; i < checkboxes.length; i++)
+      checkboxes[i].checked = !isEmpty(partitionKeys);
```

Comment:
what's happening here? seems not v react-y to be setting the element prop like this rather than via state?

Reply:
So the issue is that when a state consists of an array or object in general, react doesn't work great out of the box because you cannot control mutations as you would with a flattened state.
Let's take the example of:
```
const [foo, setFoo] = useState([]);
foo.push("c");
setFoo(foo);
```
In this case, because `oldFoo === newFoo` the `setFoo` won't trigger a re-render.

So in this particular case, I am basically:
1) using a reducer to set the initial state value (with the reset) and making sure the frontend reflects that (checking/unchecking the checkboxes). 
2) Once you make changes in the UI (you check/uncheck the boxes) that updates the state (so, basically unidirectional)
3) If you unselect a table or you select a different one, I recall the reset because I want all the partition keys to be selected as default

Reply:
you shouldn't mutate foo directly though? your example should be:
```
const [foo, setFoo] = useState([]);
setFoo(foo.concat(["c"]);
```

Then checked state on the checkbox should be set by `foo.indexOf("c") !== -1`. onChange handler for the checkbox would be something like: 
```
<Checkbox onChange={toggleCheckbox("c")} ..../>

const toggleCheckbox = val => e => {
  const isChecked = event.target.checked;
  setFoo(isChecked
    ? foo.filter(i => i !== val))
    : foo.concat([val])
  )
}
```

Reply:
Ok so I think this is clear, the change handler you suggest is `useState` based and I am using a reducer, and I think I prefer the reducer but that doesn't make such difference.

I think the ugliness consists on the fact the checkboxes didn't have a `checked` property and I was setting the initial state by checking the ID and selecting all via JS for Partition Keys and empty for columns; but I changed that now.
I am not sure why I did this when working on the columns, but I think I found a cleaner way now if you look at the new diff:
1. I have a reducer that I can use for any multi-value based array (which I have in fact moved to utils)
2. For both the `ColumnsViewer` and `PartitionKeysViewer` I am actually now using a `checked` property to get the initial state from state now
3. I have a `reset` function that allows me to set the defaults when I change the selected table (all for PKs, none for columns)
4. As you can see from the diff, I was before resetting the defaults in the `onChange` handler for the db/table dropdowns, but now I moved it to a `useEffect` as I think that's cleaner and works well.

Let me know what do you think of it now, I think it's much better and full react-y :D now